app/src/main/java/de/westnordost/streetcomplete/data/user/UserModule.kt

```diff
@@ -1,34 +1,31 @@
 package de.westnordost.streetcomplete.data.user
 
-import oauth.signpost.OAuthConsumer
-import oauth.signpost.OAuthProvider
-import org.koin.core.qualifier.named
 import org.koin.dsl.module
-import se.akerfeldt.okhttp.signpost.OkHttpOAuthConsumer
-import se.akerfeldt.okhttp.signpost.OkHttpOAuthProvider
 
-private const val BASE_OAUTH_URL = "https://www.openstreetmap.org/oauth/"
-private const val CONSUMER_KEY = "NV4cEoqQ94Kuoowh8qGJvUJLnbts40WiNykyeC1T"
-private const val CONSUMER_SECRET = "r68v1Bd7RewTixAp0dMdCwn3w5iQvmpk4HlJcH2y"
-private const val CALLBACK_SCHEME = "streetcomplete"
-private const val CALLBACK_HOST = "oauth"
+const val OAUTH2_TOKEN_URL = "https://www.openstreetmap.org/oauth2/token"
+const val OAUTH2_AUTHORIZATION_URL = "https://www.openstreetmap.org/oauth2/authorize"
```

Comment:
This wasn't present when I wrote the initial implementation for JOSM, but the OSM website now supports [rfc8414](https://datatracker.ietf.org/doc/html/rfc8414) (see https://www.openstreetmap.org/.well-known/oauth-authorization-server ).

If I were writing an implementation today, I _would not_ hardcode the OSM OAuth URLs. And I'm kind of planning on removing the hardcoded OAuth URLs in JOSM.

Reply:
But why would you not hardcode them? I get that when writing a generic OAuth client, it would be very wise to support rfc8414 but since this only speaks with OSM, it's basically one hardcoded URL (https://www.openstreetmap.org) vs two (auth url + token url).

(Also, at the time of implementation, the testing instance didn't have the .well-known file. Not sure if it changed.)

Reply:
> Also, at the time of implementation, the testing instance didn't have the .well-known file. Not sure if it changed.

The testing instance does have a .well-known file now ( https://api06.dev.openstreetmap.org/.well-known/oauth-authorization-server ).

> But why would you not hardcode them?

As a general principle, I try to avoid hardcoding things that can be obtained through standardized means (assuming the standard is reasonable). As an example, website <foo> might decide that they want to move the `/oauth2` endpoints to `/oauth` when they remove OAuth 1.0 support. I _don't_ think OSM will do that, so it probably isn't a "big" issue.

Reply:
I see, thanks!

By the way, https://github.com/openstreetmap/openstreetmap-website/issues/4360 may affect JOSM implementation, too:

According to the specification, a `scope` parameter is returned from the redirect URI which contains the actually granted scopes by the user. Meaning, according to the specification, a user may be allowed to not grant all permissions of what has been requested by the client. The app can/should handle that somehow, either rejecting the access token if it does not have all required scopes granted and/or dealing with the missing permission in the UI.
(That is, if OSM website maintainers decide that they will consider making use of this in the future. But I understand you like future-proofing JOSM for possible RFC compliant changes to the OAuth 2 API on OSM)